qiskit/algorithms/time_evolvers/trotterization/trotter_qrte.py

```diff
@@ -165,16 +163,24 @@ def evolve(self, evolution_problem: TimeEvolutionProblem) -> TimeEvolutionResult
                 "The time evolution problem contained ``aux_operators`` but no estimator was "
                 "provided. The algorithm continues without calculating these quantities. "
             )
+
+        # ensure the hamiltonian is a sparse pauli op
         hamiltonian = evolution_problem.hamiltonian
         if not isinstance(hamiltonian, (Pauli, PauliSumOp, SparsePauliOp)):
             raise ValueError(
                 f"TrotterQRTE only accepts Pauli | PauliSumOp, {type(hamiltonian)} provided."
             )
+        if isinstance(hamiltonian, PauliSumOp):
+            hamiltonian = hamiltonian.primitive
```

Comment:
Should we multiply by the coefficient just in case it is not `1`?

```suggestion
            hamiltonian = hamiltonian.primitive * hamiltonian.coeff
```